### PR TITLE
CNV-33095: add volumesnapshot option in the add bootable volume modal

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Add the VoumeSnapshot option in the bootable volume.

With this option, we'll create a `DataSource` with `spec.source.snapshot`
UI is the same as the PVC option.

## 🎥 Demo

Screenshot from 2023-10-09 10-58-23
Screenshot from 2023-10-09 10-58-04
Screenshot from 2023-10-09 10-58-01

### CODE REVIEW

File: src/utils/components/SelectSnapshot/SelectSnapshot.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  );
+
+  const snapshotNames = useMemo(
+    () =>
+      snapshots?.map((snapshot) => snapshot?.metadata?.name)?.sort((a, b) => a?.localeCompare(b)),
+    [snapshots],
+  );
+
+  return (
+    <div>
```
Comment (metalice) [inline src/utils/components/SelectSnapshot/SelectSnapshot.tsx:38]: div or fragment?

File: src/utils/components/SelectSnapshot/SelectSnapshotNamespace.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+type SelectSnapshotNamespaceProps = {
+  isDisabled?: boolean;
+  onChange: Dispatch<SetStateAction<string>>;
+  projectsLoaded: boolean;
+  projectsName: string[];
+  selectedProject: string;
+};
+
+export const FilterSnapshotSelect = (options: string[]) => {
```
Comment (metalice) [inline src/utils/components/SelectSnapshot/SelectSnapshotNamespace.tsx:17]: seems to me like its repeating with the file above

Comment (upalatucci) [reply src/utils/components/SelectSnapshot/SelectSnapshotNamespace.tsx:17]: Yes, I didn't notice. I'll do a common utils.

File: src/utils/components/SelectSnapshot/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    let newOptions = options;
+
+    if (value) {
+      const regex = new RegExp(value, 'i');
+      newOptions = options.filter((namespace) => regex.test(namespace));
+    }
+
+    return newOptions.map((namespace) => (
+      <SelectOption key={namespace} value={namespace} />
+    )) as ReactElement[];
```
Comment (metalice) [inline src/utils/components/SelectSnapshot/utils.tsx:16]: hmm returning elements from util function isnt recommended, what about return the array and in the consuming component just map over it? like we are doing with dropdown items or other select items?

File: src/utils/components/SelectSnapshot/SelectSnapshotNamespace.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import React, { Dispatch, FC, SetStateAction, useCallback, useState } from 'react';
```
Comment (metalice) [inline src/utils/components/SelectSnapshot/SelectSnapshotNamespace.tsx:1]: cant use a generic component for name and namespace? they are very similar

File: src/utils/components/SelectSnapshot/useSnapshots.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+type UseSnapshotsReturnType = {
+  error: Error;
+  projectsLoaded: boolean;
+  projectsNames: string[];
+  snapshots: VolumeSnapshotKind[];
+  snapshotsLoaded: boolean;
+};
+
+export type VolumeSnapshotKind = K8sResourceCommon & {
```
Comment (metalice) [inline src/utils/components/SelectSnapshot/useSnapshots.ts:19]: if u are exporting a type better to put in a new tyhpes.ts near util, other components can consume it not from a hook file, just make it more organize

**Pull request merged**